File: Top 100 Codes/02-Working with Numbers/20-Find the prime numbers between 1 to 100/Findtheprimenumbersbetween1to100.py
# Function to check if a number is prime
def is_prime(num):
    if num <= 1:
        return False
    for i in range(2, int(num**0.5) + 1):  # Check divisibility up to sqrt(num)
        if num % i == 0:
            return False
    return True

# Find and print prime numbers between 1 and 100
print("Prime numbers between 1 and 100 are:", end=" ")
for num in range(1, 101):
    if is_prime(num):
        print(num, end=" ")

print()  # For newline after output

File: Top 100 Codes/02-Working with Numbers/20-Find the prime numbers between 1 to 100/Findtheprimenumbersbetween1to100.cpp
#include <iostream>
using namespace std;

int main() {
    // Display prime numbers between 1 and 100
    cout << "Prime numbers between 1 and 100 are: ";

    for (int num = 1; num <= 100; num++) {
        if (num <= 1) continue; // Skip numbers less than or equal to 1
        
        bool isPrime = true;

        // Check divisibility from 2 to sqrt(num)
        for (int i = 2; i * i <= num; i++) {
            if (num % i == 0) {
                isPrime = false;
                break;
            }
        }

        if (isPrime) {
            cout << num << " ";
        }
    }

    cout << endl;
    return 0;
}
